backends/bmv2/jsonconverter.cpp

```diff
@@ -112,6 +112,26 @@ class SharedActionSelectorCheck : public Inspector {
     using Input = std::vector<const IR::Expression *>;
     std::map<const IR::Declaration_Instance *, Input> selector_input_map{};
 
+  static bool checkSameKeyExpr(const IR::Expression* expr0, const IR::Expression* expr1) {
+      if (expr0->node_type_name() != expr1->node_type_name())
+          return false;
+      if (auto pe0 = expr0->to<IR::PathExpression>()) {
+          auto pe1 = expr1->to<IR::PathExpression>();
+          return pe0->path->name == pe1->path->name &&
+              pe0->path->absolute == pe1->path->absolute;
+      } else if (auto mem0 = expr0->to<IR::Member>()) {
+          auto mem1 = expr1->to<IR::Member>();
+          return checkSameKeyExpr(mem0->expr, mem1->expr) && mem0->member == mem1->member;
+      } else if (auto c0 = expr0->to<IR::Constant>()) {
```

Comment:
Here you may use to<IR::Literal>; this will handle booleans and enums, in case these are legal in key expressions.
For these cases you can probably use the natural equality operator expr0->operator==(*expr1), since these are leaf expressions.